pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -120,6 +168,22 @@ func toRuntimeProtocol(protocol api.Protocol) runtimeApi.Protocol {
 	return runtimeApi.Protocol_TCP
 }
 
+// toKubeContainer converts runtimeApi.Container to kubecontainer.Container.
+func (m *kubeGenericRuntimeManager) toKubeContainer(c *runtimeApi.Container) (*kubecontainer.Container, error) {
+	if c == nil || c.Id == nil || c.Name == nil || c.Image == nil || c.State == nil {
+		return nil, fmt.Errorf("unable to convert a nil pointer to a runtime container")
+	}
+
+	_, _, _, cName, hash, _ := parseContainerName(c.GetName())
```

Comment:
Even for current docker implementation, we already have hash labels.
Should we get hash from labels here?